### DISCUSSION TITLE
Feature Request: bookmark folder locations

### DISCUSSION THREAD

**Discussion by mister-bum** [puddletag/puddletag#661] [Ideas]
Having used MP3tag for nearly 20 years I have to say, that puddletag already fulfills like 95% of my needs - thanks a lot for this to everyone involved !!!

However one feature that I miss (or haven't found yet) is the function to have (hidden) folders bookmarked as favorite locations. I am mainly working on two different directories (one hidden and the other non-hidden), and a bookmark feature would make it easier to switch between those directories.

Working with symlinks and the "load last folder on startup" option could be a workaround, but this is not really desirable for reasons that are probably of no interest here (I can elaborate if requested).

My coding skills are definitely limited but then again not entirely blank. So if I can contribute in some way, I'd be willing to try.

puddletag 2.0.1, kubuntu 20.04

**Comment by sandrotosi** [comment]
> Having used MP3tag for nearly 20 years I have to say, that puddletag already fulfills like 95% of my needs - thanks a lot for this to everyone involved !!!
> 
> However one feature that I miss (or haven't found yet) is the function to have (hidden) folders bookmarked as favorite locations. I am mainly working on two different directories (one hidden and the other non-hidden), and a bookmark feature would make it easier to switch between those directories.
> 
> Working with symlinks and the "load last folder on startup" option could be a workaround, but this is not really desirable for reasons that are probably of no interest here (I can elaborate if requested).

can you provide more information about what an hidden directory is in the context of mp3tag / what would look like in puddletag?

> My coding skills are definitely limited but then again not entirely blank. So if I can contribute in some way, I'd be willing to try.

we are definitely happy if you want to give it a try and open a PR for it :)

**Reply by mister-bum** [reply]
> can you provide more information about what an hidden directory is in the context of mp3tag / what would look like in puddletag?

First of all, the hidden directory is something that I have never used before (i.e. with MP3tag). This is to point out that my post was aimed at the bookmark feature, and not so much about the hidden directory. Sorry if the mentioned "hidden directory" distracted from that.

But I  will of course explain my idea:
I have been migrating from Windows to Ubuntu during the last year. Basically I am very happy about this step, but I still haven't found a good and reliable audio player yet.
As some of the tested players can also edit the metadata and/or file structure, I do not want to give them rw access to my collection, especially as I have read a few very scary reports about features like auto-tagging or re-organisation that have blown a whole collection to pieces. So I made up the following concept:

My music collection is stored on a NAS, which I have mounted using cifs as a read-only mount to be used by any software who does not necessarily needs write permission, e.g.
_/media/user/MP3_Collection_

However for programs like puddletag, I mount the collection a second time, but this time time with rw permission. And THIS mount point (precisely the folder) is hidden, so that cannot be used accidentally, e.g.
_/media/user/.MP3_Collection._

In addition to the NAS, I have of course a local working directory, for splitting and editing of new audio files. These files are then tagged and transferred by puddletag, using the hidden mount point - and this already works like a breeze :-)

But it would be handy, if I could just save the paths to those two directories (or maybe a few more) as bookmarks, especially as it takes a few extra steps to navigate to the hidden folder (as it is hidden).
